### PR TITLE
Fix #1853: App can't be deleted because it "does not exist", although /apps still returns it

### LINKED ISSUE #1853: App can't be deleted because it "does not exist", although /apps still returns it
I was running an instance of Marathon v0.9, then stopped it, then started an instance of v0.10.0-RC3 against the same Mesos instance v0.22.1.

When I GET the `/apps` endpoint, i see:

```json
{
  "apps": [
    {
      "id": "/foosball",
      "cmd": "/usr/sbin/nginx",
      "args": null,
      "user": null,
      "env": {},
      "instances": 0,
      "cpus": 0.1,
      "mem": 32,
      "disk": 0,
      "executor": "",
      "constraints": [],
      "uris": [],
      "storeUrls": [],
      "ports": [],
      "requirePorts": false,
      "backoffSeconds": 1,
      "backoffFactor": 1.15,
      "maxLaunchDelaySeconds": 3600,
      "container": {
        "type": "DOCKER",
        "volumes": [],
        "docker": {
          "image": "philipnrmn/mesosphere-foosball",
          "portMappings": [
            {
              "containerPort": 80,
              "hostPort": 0,
              "servicePort": 0,
              "protocol": "tcp"
            }
          ],
          "privileged": false,
          "parameters": [],
          "forcePullImage": false
        }
      },
      "healthChecks": [],
      "dependencies": [],
      "upgradeStrategy": {
        "minimumHealthCapacity": 1,
        "maximumOverCapacity": 1
      },
      "labels": {},
      "acceptedResourceRoles": null,
      "version": "2015-07-20T15:57:34.009Z",
      "tasksStaged": 0,
      "tasksRunning": 0,
      "tasksHealthy": 0,
      "tasksUnhealthy": 0,
      "deployments": []
    }
  ]
}
```

but if when I try to delete it, I get a 404: 

```
curl -X DELETE localhost:8080/v2/apps/foosball
{"message":"App '/foosball' does not exist"}
```

In the marathon logs, I see:

```
WARN  (mesosphere.marathon.api.MarathonExceptionMapper$$EnhancerByGuice$$fa19ac1f:30)
mesosphere.marathon.UnknownAppException: App '/foosball' does not exist
	at mesosphere.marathon.api.v2.AppsResource$$anonfun$mesosphere$marathon$api$v2$AppsResource$$deleteApp$1$2.apply(AppsResource.scala:195)
	at mesosphere.marathon.api.v2.AppsResource$$anonfun$mesosphere$marathon$api$v2$AppsResource$$deleteApp$1$2.apply(AppsResource.scala:195)
	at scala.Option.getOrElse(Option.scala:121)
	at mesosphere.marathon.api.v2.AppsResource.mesosphere$marathon$api$v2$AppsResource$$deleteApp$1(AppsResource.scala:195)
	at mesosphere.marathon.api.v2.AppsResource$$anonfun$16.apply(AppsResource.scala:197)
	at mesosphere.marathon.api.v2.AppsResource$$anonfun$16.apply(AppsResource.scala:197)
	at mesosphere.marathon.state.Group$$anonfun$update$1.apply(Group.scala:59)
	at mesosphere.marathon.state.Group$$anonfun$update$1.apply(Group.scala:58)
	at mesosphere.marathon.state.Group.update(Group.scala:72)
	at mesosphere.marathon.state.Group.update(Group.scala:58)
	at mesosphere.marathon.state.GroupManager$$anonfun$update$1.apply(GroupManager.scala:109)
	at mesosphere.marathon.state.GroupManager$$anonfun$update$1.apply(GroupManager.scala:109)
	at mesosphere.marathon.state.GroupManager$$anonfun$upgrade$1$$anonfun$2.apply(GroupManager.scala:147)
	at mesosphere.marathon.state.GroupManager$$anonfun$upgrade$1$$anonfun$2.apply(GroupManager.scala:146)
	at scala.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:251)
	at scala.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:249)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

Let me know if there is anything else I can attach to this issue for inspection.

### PR DESCRIPTION
**Pull request by aquamatthias**
This fix uses the groupManager to list applications instead of the AppsRepository.

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala [hunk 1]
```diff
@@ -112,6 +112,7 @@ class MarathonSchedulerService @Inject() (
   def cancelDeployment(id: String): Unit =
     schedulerActor ! CancelDeployment(id)
 
+  @deprecated("Use GroupManager.root.transitiveApps", "0.10.0")
```
Comment (kolloch) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala]: why not delete this method?

It is slightly strange to use the AppRepository for versioning and nothing else, but hey!

**Comment by kolloch** [comment]
This looks good but I think we have to clarify some decisions: What do we use the `AppRepository` for? What do we use the `GroupRepository` for? It would be nice if you documented that in the source code. 

So my working theory is that we only use the `AppRepository` for storing versioned `AppDefinition`s.

Your explanation might already answer my next questions:

* We use `appRepository.apps()` for reconciliation. Is that correct?
* Shouldn't we store new versioned `AppDefinition`s immediately at the start of a deployment?

**Comment by kolloch** [comment]
LGTM! for now ;)

Remove the deprecated method, squash to one commit and I'll merge.

**Comment by aquamatthias** [comment]
Added #1898 for cleaning things up

**Comment by kolloch** [comment]
:+1:

**Pull request merged**